### DISCUSSION TITLE
parallel SGP training

### DISCUSSION THREAD

**Discussion by TSMtang** [mir-group/flare#336] [Q&A] [answered]
Dear Flare developers,

I've been using FLARE for training force field. I noticed that when using Sparse Gaussian Process regression (gp: SGP_Wrapper), there is no option to setup the number of CPUs (e.g. n_cpus: 10 is not working). I wonder if there is any way I can use SGP_Wrapper with multiple processors?

Thank you in advance
Shengming Tang

**Comment by YuuuXie** [comment, marked as answer]
Hi @TSMtang , in `SGP_Wrapper` you don't need to specify number of CPUs. Instead you can set the environmental variable:
```
export OMP_NUM_THREADS=<number_of_cpus>
```
before you run your training. The number of CPUs has to be equal or smaller than the number on a single node, since we use OpenMP to parallelize the SGP
